Automattic/wp-calypso
# Add Professional Email to cart in signup

#### Changes proposed in this Pull Request

* This PR updates the function for adding plans to the cart to also check for a Professional Email subscription and add both items at the same time.
* The PR also includes some small changes to `SignupFlowController`:
  - We now supply optional dependencies to the completion step (as this allows us to pull in the `emailItem` in the plan step)
  - We use a local helper function for building dependency data (and no longer use `get` from Lodash)

#### Testing instructions

* Open this branch locally or on the live branch (see [this comment](https://github.com/Automattic/wp-calypso/pull/54743#issuecomment-884233089) for the current link)
* Navigate to `/start?flags=signup/professional-email-step`
* Pick a paid domain on the domains step
* Click on "Add" on the emails step
* Pick a paid plan on the plans step
* Verify that checkout is then loaded after the progress screen, and that all three products are listed

Try the above with monthly and annual plans; also try adding a domain without a plan (if that's allowed?), try with free subdomain and with a paid plan (i.e. no domain or email), and make sure things still work.

#### Screenshots

##### Monthly plan with domain and Professional Email

<img width="1666" alt="Monthly Business Plan with domain and Professional Email" src="https://user-images.githubusercontent.com/3376401/126526967-46155ad0-e2af-4f01-be3f-7653c7a1e2c4.png">

##### Annual plan with domain and Professional Email

<img width="1666" alt="Annual Business Plan with domain and Professional Email" src="https://user-images.githubusercontent.com/3376401/126526970-ec295ea5-cf85-4b4a-964b-e8bf89408d72.png">

Related to #54703
